### PR TITLE
Address Early Bean Initialization

### PR DESCRIPTION
**Pull request by lukasniemeier-zalando**
Fixes GH-391.

See commits for details.

### CODE REVIEW

File: opentracing-jdbc/opentracing-jdbc-autoconfigure/src/main/java/org/zalando/opentracing/jdbc/autoconfigure/TracingProcessor.java [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
+import org.springframework.beans.factory.BeanFactory;
 import org.zalando.opentracing.jdbc.DataSourceTracer;
 
 import javax.sql.DataSource;
 
 final class TracingProcessor extends ForwardingProcessor {
 
-    public TracingProcessor(final DataSourceTracer tracer) {
-        super(new ConditionalProxyProcessor<>(DataSource.class, tracer::trace));
+    public TracingProcessor(final BeanFactory beanFactory) {
```
Comment (lukasniemeier-zalando) [inline opentracing-jdbc/opentracing-jdbc-autoconfigure/src/main/java/org/zalando/opentracing/jdbc/autoconfigure/TracingProcessor.java]: looking at other post processors, apparently the "idiomatic" way is to mark the processor as `BeanFactoryAware`, but I wanted to keep the changes minimal not break up `ConditionalProxyProcessor`.

**Comment by whiskeysierra** [comment]
:+1:

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Comment by whiskeysierra** [comment]
:+1:

**Pull request merged**